src/image/PhotoCalib.cc

```diff
@@ -465,38 +544,68 @@ double PhotoCalib::evaluate(lsst::geom::Point<double, 2> const &point) const {
         return _calibration->evaluate(point);
 }
 
+ndarray::Array<double, 1> PhotoCalib::evaluateArray(ndarray::Array<double, 1> xx,
+                                                    ndarray::Array<double, 1> yy) const {
+    if (_isConstant) {
+        ndarray::Array<double, 1> result = ndarray::allocate(ndarray::makeVector(xx.size()));
+        result.deep() = _calibrationMean;
+        return result;
+    } else {
+        return _calibration->evaluate(xx, yy);
+    }
+}
+
+ndarray::Array<double, 1> PhotoCalib::evaluateCatalog(afw::table::SourceCatalog const &sourceCatalog) const {
+    ndarray::Array<double, 1> xx = ndarray::allocate(ndarray::makeVector(sourceCatalog.size()));
+    ndarray::Array<double, 1> yy = ndarray::allocate(ndarray::makeVector(sourceCatalog.size()));
+    size_t i = 0;
+    for (auto const &rec : sourceCatalog) {
+        auto point = rec.getCentroid();
+        xx[i] = point.getX();
+        yy[i] = point.getY();
+        ++i;
+    }
+    return evaluateArray(xx, yy);
+}
+
 void PhotoCalib::instFluxToNanojanskyArray(afw::table::SourceCatalog const &sourceCatalog,
                                            std::string const &instFluxField,
                                            ndarray::Array<double, 2, 2> result) const {
-    double instFlux, instFluxErr, nanojansky, calibration;
+    double instFlux, instFluxErr, nanojansky;  //, calibration;
```

Comment:
It's fine (i.e. just as efficient) and probably slightly better form to declare these on first use, even though that puts them inside the loop - it's always good to avoid uninitialized variables, and for stack variables of trivial types like `double` you don't need to worry about the cost of construction being different from the cost of assignment.